docsrc/allblocks.nim

```diff
@@ -1,40 +1,126 @@
 import nimoji
 import nimib
+import nimpy
+import std/[math, strutils, strformat]
 
 nbInit
 
+var nbToc: NbBlock
+
+template addToc =
+  newNbBlock("nbText", false, nb, nbToc, ""):
+    nbToc.output = "## Table of Contents:\n\n"
+
+template nbCodeBlock(name:string) =
+  let anchorName = name.toLower.replace(" ", "-")
+  nbText "<a name = \"" & anchorName & "\"></a>\n### " & name & "\n\n---"
+  # see below, but any number works for a numbered list
+  nbToc.output.add "1. <a href=\"#" & anchorName & "\">" & name & "</a>\n"
+
 nbText: """
 > This nimib document provides a brief description and example for 
 > all blocks in nimib.
 
 :warning: **This document is not finished as of now.**
 """.emojize
 
-nbText: """
-### nbCodeSkip
+addToc()
 
+nbCodeBlock:"nbCodeSkip"
```

Comment:
```suggestion
nbCodeBlock: "nbCodeSkip"
```
looks a bit weird to have no space between `:` and the string (also in all other blocks below)